pkg/selinux/setroubleshoot-client.js

```diff
@@ -107,69 +107,56 @@
-                            lastSeen: result[6] / 1000,
-                            level: result[7],
-                        };
-                        // cleanup analysis
-                        details.pluginAnalysis = details.pluginAnalysis.map(itm => ({
-                            ifText: itm[0],
-                            thenText: itm[1],
-                            doText: itm[2],
-                            analysisId: itm[3],
-                            fixable: itm[4],
-                            reportBug: itm[5],
-                        }));
-                        resolve(details);
-                    })
-                    .catch(ex => {
-                        console.warn("Unable to get alert for id " + localId);
-                        console.warn(ex);
-                        reject(new Error(cockpit.format(_("Unable to get alert: $0"), localId)));
-                    });
-        });
-    };
+    client.getAlert = localId => client.proxy.call("get_alert", [localId])
+            .then(result => {
+                const details = {
+                    localId: result[0],
+                    summary: result[1],
+                    reportCount: result[2],
+                    auditEvent: result[3],
+                    pluginAnalysis: result[4],
+                    firstSeen: result[5] / 1000,
+                    lastSeen: result[6] / 1000,
+                    level: result[7],
+                };
+                // cleanup analysis
+                details.pluginAnalysis = details.pluginAnalysis.map(itm => ({
+                    ifText: itm[0],
+                    thenText: itm[1],
+                    doText: itm[2],
+                    analysisId: itm[3],
+                    fixable: itm[4],
+                    reportBug: itm[5],
+                }));
+                return details;
+            })
+            .catch(ex => {
+                console.warn("Unable to get alert for id", localId, ":", JSON.stringify(ex));
+                return Promise.reject(new Error(`Unable to get alert for id ${localId}: ${ex.toString()}`));
+            });
 
     /* Run a fix via SetroubleshootFixit
        The analysisId is given as part of pluginAnalysis entries in alert details
      */
-    client.runFix = function(alertId, analysisId) {
-        return new Promise((resolve, reject) => {
-            client.proxyFixit.call("run_fix", [alertId, analysisId])
-                    .then(result => resolve(result[0]))
-                    .catch(ex => reject(new Error(cockpit.format(_("Unable to run fix: $0"), ex))));
-        });
-    };
+    client.runFix = (alertId, analysisId) => client.proxyFixit.call("run_fix", [alertId, analysisId])
+            .then(result => result[0])
+            .catch(ex => {
+                console.warn("Unable to run fix:", JSON.stringify(ex));
+                return Promise.reject(new Error(cockpit.format(_("Unable to run fix: $0"), ex.toString())));
```

Comment:
These 3 added lines are not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-18670-20230419-114356-a9ad7ebb-fedora-37-devel//Coverage/lcov/github-pr.diff.gcov.html)